src/Psy/TabCompletion/Matcher/ObjectMethodDefaultParametersMatcher.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Psy\TabCompletion\Matcher;
+
+class ObjectMethodDefaultParametersMatcher extends AbstractContextAwareMatcher
+{
+    public function getMatches(array $tokens, array $info = array())
+    {
+        $openBracket = array_pop($tokens);
+        $functionName = array_pop($tokens);
+        $methodOperator = array_pop($tokens);
+
+        $objectToken = array_pop($tokens);
+        if (!is_array($objectToken)) {
+            return array();
+        }
+
+        $objectName = str_replace('$', '', $objectToken[1]);
+
+        try {
+            $object = $this->getVariable($objectName);
+            $reflection = new \ReflectionObject($object);
+        } catch (InvalidArgumentException $e) {
+            return array();
+        } catch (\ReflectionException $e) {
+            return array();
+        }
+
+        $methods = $reflection->getMethods();
+
+        foreach ($methods as $method) {
+            if ($method->getName() === $functionName[1]) {
+                $parameterString = $this->extractParameterString($method);
+
+                if (empty($parameterString)) {
+                    return array();
+                }
+
+                return array($parameterString);
+            }
+        }
+
+        return array();
+    }
+
+    /**
+     * @param \ReflectionMethod $method
+     *
+     * @return string
+     */
+    private function extractParameterString($method)
```

Comment:
please use types for arguments where possible.

Reply:
That's a good point. I'm used to putting type hinting everywhere, but I removed it all when I realised that wouldn't work all the way back to PHP 5.3. I was a little overzealous.